src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -89,13 +89,31 @@ def register_attributes_argument(scope, name, attr_class, create=False):
 
 register_attributes_argument('keyvault key set-attributes', 'key', KeyAttributes)
 
-register_cli_argument('keyvault secret', 'secret_version', options_list=('--version', '-v'), help='The secret version. If omitted, uses the latest version.')
+register_cli_argument('keyvault secret', 'secret_version', options_list=('--version', '-v'), help='The secret version. If omitted, uses the latest version.', default='', required=False)
 
 register_attributes_argument('keyvault secret set', 'secret', SecretAttributes, create=True)
 register_attributes_argument('keyvault secret set-attributes', 'secret', SecretAttributes)
 
-register_cli_argument('keyvault certificate', 'certificate_version', options_list=('--version', '-v'), help='The certificate version. If omitted, uses the latest version.')
-
+register_cli_argument('keyvault certificate', 'certificate_version', options_list=('--version', '-v'), help='The certificate version. If omitted, uses the latest version.', default='', required=False)
+register_attributes_argument('keyvault certificate create', 'certificate', CertificateAttributes, True)
+register_attributes_argument('keyvault certificate set-attributes', 'certificate', CertificateAttributes)
 for item in ['create', 'set-attributes']:
-    register_attributes_argument('keyvault certificate {}'.format(item), 'certificate', CertificateAttributes, item == 'create')
-    register_cli_argument('keyvault certificate {}'.format(item), 'certificate_policy', options_list=('--policy', '-p'), help='JSON encoded policy defintion. Use @{file} to load from a file.', type=json.loads)
+    register_cli_argument('keyvault certificate {}'.format(item), 'certificate_policy', options_list=('--policy', '-p'), help='JSON encoded policy defintion. Use @{file} to load from a file.', type=json_string_type)
+
+register_cli_argument('keyvault certificate contact', 'contact_email', options_list=('--email',), help='Contact e-mail address. Must be unique.')
```

Comment:
Contacts are tied to the keyvault, so it must unique in the vault. I'll clarify that (since I would assume it would be specific to the certificate!)